sql-plugin/src/main/scala/org/apache/spark/sql/execution/joins/rapids/CpuSortMergeJoinFallback.scala

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.joins.rapids
+
+import com.nvidia.spark.rapids.{ColumnarToRowIterator, GpuOutOfCoreSortIterator, RowToColumnarIterator}
+import com.nvidia.spark.rapids.RapidsBuffer.SpillCallback
+
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.expressions.{Ascending, Attribute, Expression, JoinedRow, Predicate, Projection, RowOrdering, SortOrder, UnsafeProjection, UnsafeRow}
+import org.apache.spark.sql.catalyst.plans.{InnerLike, JoinType}
+import org.apache.spark.sql.execution.{ExternalAppendOnlyUnsafeRowArray, RowIterator}
+import org.apache.spark.sql.execution.joins.SortMergeJoinScanner
+import org.apache.spark.sql.vectorized.ColumnarBatch
+
+class CpuSortMergeJoinFallback(
+    val joinType: JoinType,
```

Comment:
Wow, this is a lot of parameters with a lot of the same types.  Keeping the call sites ordered properly will be challenging, good thing there aren't that many sites.

Should this take the `CpuJoinInfo` directly rather than as separate parameters?